common/src/BasePlugin.php

```diff
@@ -116,30 +116,59 @@ public function getProductMetaKeys(): array {
             WHERE
                 p.post_type = 'product'
                 AND pm.meta_key <> ''
+                AND pm.meta_value <> ''
         ");
         return array_map(
             function ($value) {
                 return [
                     'type' => 'meta_key',
-                    'name' => $value
+                    'name' => $value,
+                    'example_value' => substr($this->getMetaValue($value), 0, 15),
                 ];
             },
             $meta_keys
         );
     }
 
-    public function getProductKeys() {
+    public function getSelectOptions(string $selected_key, bool $suggested = false) {
+        $options = [];
+        foreach ($this->getProductKeys() as $key) {
+            if ($key['suggested'] != $suggested) {
+                continue;
+            }
+            $options[] = '<option value="' . $key['option_value'] . '" '
+                . ($key['option_value'] == $selected_key ? 'selected' : '') . '>'
+                . $key['label']
+                . '</option>';
+        }
+        return $options;
+    }
+
+    private function getProductKeys(): array {
         $custom_keys = array_merge($this->getProductMetaKeys(), $this->getCustomAttributes());
         return array_map(function ($value) {
             return [
                 'option_value' => $value['type'] . htmlentities($value['name']),
-                'label' => htmlentities($value['name']) . ' (' . $value['type'] . ')',
+                'label' => htmlentities($value['name']) . ' (e.g. "' . $value['example_value'] . '")',
+                'suggested' => $this->isValidGtin($value['example_value']),
             ];
         },
             $custom_keys
         );
     }
 
+    private function getMetaValue(string $meta_key) {
+        global $wpdb;
+        $sql = "
+            SELECT meta.meta_value
+            FROM {$wpdb->postmeta} meta
+            WHERE meta.meta_key = '{$meta_key}'
```

Comment:
You must escape `$meta_key`! There's no guarantee someone won't make a meta key that contains quotes.  So use [WordPress' functionality for preparing statements with `%s`](https://developer.wordpress.org/reference/classes/wpdb/#protect-queries-against-sql-injection-attacks).